manager/assets/modext/util/utilities.js

```diff
@@ -570,8 +570,62 @@ MODx.util.url = {
             newUrl = newUrl.toString().replace(/%2F/g, '/');
             window.history.replaceState(stateData, document.title, newUrl);
         }
+    },
+    /**
+     * @property {Function} clearParam - Clears a single url parameter;
+     * the param name is derived from the calling filter's component itemId.
+     *
+     * @param {Object|String} reference - The filter's Ext component or the parameter name
+     * @param {Boolean} referenceIsComponent - Set to true if deriving parameter from a filter's Ext component, false
+     * @param {Object} stateData - Optional data to be used in subsequent url processing
+     */
+    clearParam: function(reference, referenceIsComponent = true, stateData = {}) {
+        if (typeof window.history.replaceState !== 'undefined') {
+            const urlParts = window.location.href.split('?'),
+                  removeParamName = referenceIsComponent ? this.getParamNameFromCmp(reference) : reference.trim(),
+                  regex = new RegExp(`${removeParamName}=[^&]+`, 'i')
+            ;
+            let params = urlParts[1].replace(regex, '').replace('&&', '&');
+            if (params.endsWith('&')) {
+                params = params.substr(0, params.length - 1);
+            }
+            let newUrl = new URL(`${urlParts[0]}?${params}`);
+            newUrl = newUrl.toString().replace(/%2F/g, '/');
```

Comment:
```javascript
const url = new URL(window.location.href)
url.searchParams.delete(removeParamName)
```

Reply:
Ok, using the URL API does help to simplify it a bit. I updated the method, although it needed more than just the two lines you suggested.